llvm/lib/Debuginfod/Debuginfod.cpp

```diff
@@ -62,15 +66,23 @@ bool canUseDebuginfod() {
 }
 
 SmallVector<StringRef> getDefaultDebuginfodUrls() {
-  const char *DebuginfodUrlsEnv = std::getenv("DEBUGINFOD_URLS");
-  if (DebuginfodUrlsEnv == nullptr)
-    return SmallVector<StringRef>();
-
-  SmallVector<StringRef> DebuginfodUrls;
-  StringRef(DebuginfodUrlsEnv).split(DebuginfodUrls, " ");
+  if (!DebuginfodUrlsSet) {
+    // Only read from the environment variable if the user hasn't already
+    // set the value
+    if (const char *DebuginfodUrlsEnv = std::getenv("DEBUGINFOD_URLS"))
+      StringRef(DebuginfodUrlsEnv).split(DebuginfodUrls, " ", -1, false);
+    DebuginfodUrlsSet = true;
+  }
   return DebuginfodUrls;
 }
 
+// Set the default debuginfod URL list, override the environment variable
+void setDefaultDebuginfodUrls(SmallVector<StringRef> URLs) {
+  DebuginfodUrls.clear();
+  DebuginfodUrls.insert(DebuginfodUrls.begin(), URLs.begin(), URLs.end());
+  DebuginfodUrlsSet = true;
```

Comment:
If we use std::optional this becomes:
```
DebuginfodUrls = URLs;
```
And the std::optional will have a new value and the value will be set.

Reply:
So much cleaner!